js/ui/handler/scroll_zoom.js

```diff
@@ -97,6 +97,7 @@ ScrollZoom.prototype = {
     },
 
     _zoom: function (delta) {
+        if (delta === 0) return;
```

Comment:
are you thinking about this from a "floating point issues may cause this check to fail" point of view or a "some legitimately non-zero but still tiny value will trigger a switch to linear but not move the map" view?